### PR TITLE
Add Extra MRC save debug logs

### PR DESCRIPTION
**Pull request by Praveenhp7**
Add more debug prints to defferentiate first flash boot and fast boot.

Signed-off-by: Praveen Hp <praveen.hodagatta.pranesh@intel.com>

### CODE REVIEW

**Review by myguitar — APPROVED** [review]
Looks good to me

File: BootloaderCorePkg/Stage2/Stage2.c [hunk 1]
```diff
@@ -412,9 +412,9 @@ SecStartup (
   // Save NVS data
   NvsData = GetFspNvsDataBuffer (LdrGlobal->FspHobList, &MrcDataLen);
   if ((NvsData != NULL) && (MrcDataLen > 0)) {
-    DEBUG ((DEBUG_INFO, "Save MRC Training Data (0x%p 0x%06X) ... ", NvsData, MrcDataLen));
+    DEBUG ((DEBUG_INFO, "Save MRC Training Data (0x%p 0x%06X) ... \n", NvsData, MrcDataLen));
     SubStatus = SaveNvsData (NvsData, MrcDataLen);
-    DEBUG ((DEBUG_INFO, "%X\n", SubStatus));
+    DEBUG ((DEBUG_INFO, "SaveNvsData SubStatus: %r\n", SubStatus));
```
Comment (mauricema) [inline BootloaderCorePkg/Stage2/Stage2.c:417]: Let us move to a single line DEBUG to save the code size.

File: Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c [hunk 2]
```diff
@@ -1752,6 +1752,7 @@ SaveNvsData (
   // if they match, no need to update again.
   //
   if (CompareMem ((VOID *)(UINTN)Address, Buffer, Length) == 0) {
+    DEBUG ((DEBUG_INFO, "MRC data matches in flash, skip update!\n"));
```
Comment (mauricema) [inline Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:1755]: I would suggest to return different status code.  core code will print the status.
In this way, it saves size.

Comment (mauricema) [reply Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:1755]: No, we don't recreate new status code. Instead, try to reuse existing ones.  We actually tried to use ALREADY_STARTED status to indicate that. So if you see ALREADY_STARTED print status, you know the MRC NVS matches and saving is skipped.

File: BootloaderCorePkg/Stage2/Stage2.c (hunk 1 above)
Comment (mauricema) [reply BootloaderCorePkg/Stage2/Stage2.c:417]: I meant you have two DEBUG statements.  One before SaveNvsData() and one after.  
If you want to make it clear,  just print the full DEBUG message after the SaveNvsData call.

File: Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c (hunk 2 above)
Comment (Praveenhp7) [reply Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c]: yes, I can see "Already started" which is getting printed in core code for MRC fast boot.  thanks for suggestion
 
    First flash boot:
       MRC data successfully cached to 0x1FF000
       Save MRC Training Data (0x8D131480 0x0018E0) ...Success

    MRC fast boot:
       Save MRC Training Data (0x8D131398 0x0018E0) ...Already started

File: BootloaderCorePkg/Stage2/Stage2.c (hunk 1 above)
Comment (Praveenhp7) [reply BootloaderCorePkg/Stage2/Stage2.c]: done, combined both prints, and moved after SaveNvsData call.

**Review by mauricema — APPROVED** [review]
Looks good.

**Pull request merged**